### PR TITLE
feat: distribute inflation rewards to block authors 

### PR DESCRIPTION
**Pull request by MaciejBaj**
# Summary of changes

**_Summarize_** the description of your changes.

If it fixes a bug or creates a feature, link the issue.

# Acceptance Checklist

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

## Dependencies

- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any teams which manage the dependencies have been notified of the breaking changes

## Bug (non-breaking change which fixes an issue):

- [ ] Have you **_written tests_** to protect against future occurrences of the bug?

## Feature (non-breaking change which adds functionality)

- [ ] Have you **_seen it working_** in a development environment?
- [ ] Have you **_written tests_** for **_happy_** and **_unhappy_** paths?
- [ ] Have you implemented this feature as per the **_issue acceptance criteria_**?

### Substrate

If this change involves substrate, have you remembered:

- [ ] Storage Migration?
- [ ] RPC methods?
- [ ] Chainspec, both JSON specs & the module?
- [ ] Runtime versioning?
- [ ] Benchmarks?

## Breaking change (a feature that would cause existing functionality not to work as expected)

- [ ] Is the breaking change **_documented_**?
- [ ] Are your commits fully representative of the change?
- [ ] Has any previous functionality been **_deprecated_** and versioned?

## CI/CD

- [ ] If introducing new actions, have you **_looked at the action code_** for anything spurious?
- [ ] Have you written **_custom scripts_** for things that could be actions already on the marketplace?
- [ ] If introducing new actions, have you **_pegged a static version_**?

## Documentation Update

- [ ] Have you checked other documentation, such as the docs repository?
- [ ] If updating script documentation, have you **_tested it on both environments_**?

## Further comments

Feel free to explain in further detail your choices if this is a significant change.

### Summary by OpenAI

**Release Notes**
- `NEEDS_REVIEW`: Changes to block author selection in `mini-mock` and associated types in `mock`, `standalone`, and `t0rn-parachain` runtimes.
- `NEEDS_REVIEW`: Modification to `pallet_portal_rpc` import statement in `rpc.rs`.
- `NEEDS_REVIEW`: Removal of `PortalExecution` struct and usage in `3vm` precompile.
- `APPROVED`: Removal of unused import statement and changes to test cases in `portal/tests.rs`.
- `NEEDS_REVIEW`: Minor changes to `grandpa` import statement and `mock/lib.rs`.

> A code review done with care,  
> Ensures our system's health is fair.  
> With new structs and imports too,  
> We'll make sure our code stays true.

### CODE REVIEW

**Review by palozano — CHANGES_REQUESTED** [review]
LGTM when CI/CD is passsing (t3rn-primites cannot be found)

File: runtime/mock/src/circuit_config.rs [hunk 1]
```diff
@@ -192,6 +192,7 @@ impl pallet_rewards::Config for Runtime {
     type Event = Event;
     type ExecutorBootstrapRewards = ExecutorBootstrapRewards;
     type ExecutorInflation = ExecutorInflation;
+    type FindAuthor = ();
     type InflationDistributionPeriod = InflationDistributionPeriod;
     type OneYear = OneYear;
     type TotalInflation = TotalInflation;
```
[the remark this answers is not in the document]
Comment (MaciejBaj) [reply runtime/mock/src/circuit_config.rs:198]: thanks robot

File: runtime/t0rn-parachain/src/circuit_config.rs [hunk 2]
```diff
@@ -182,6 +182,7 @@ impl pallet_rewards::Config for Runtime {
     type Event = Event;
     type ExecutorBootstrapRewards = ExecutorBootstrapRewards;
     type ExecutorInflation = ExecutorInflation;
+    type FindAuthor = pallet_session::FindAccountFromAuthorIndex<Self, Aura>;
     type InflationDistributionPeriod = InflationDistributionPeriod;
     type OneYear = OneYear;
     type TotalInflation = TotalInflation;
```
[the remark this answers is not in the document]
Comment (MaciejBaj) [reply runtime/t0rn-parachain/src/circuit_config.rs:188]: I know, right?

File: pallets/rewards/src/lib.rs [hunk 3]
```diff
@@ -517,6 +543,51 @@ pub mod pallet {
...
+                            .iter()
+                            .filter_map(|(author, count)| {
+                                if *count == min_count {
+                                    Some(author.clone())
+                                } else {
+                                    None
+                                }
+                            })
+                            .collect();
+                        for author in least_blocks_authors {
```
Comment (AwesomeIbex) [inline pallets/rewards/src/lib.rs:569]: i think you can just use sort_by here

**Review by AwesomeIbex — COMMENTED** [review]
lgtm one quick thing to avoid manually sorting

File: pallets/rewards/src/lib.rs (hunk 3 above)
Comment (MaciejBaj) [reply pallets/rewards/src/lib.rs]: changed to use min_by https://github.com/t3rn/t3rn/pull/966/commits/000692d75a8ef05f8a929416721f25067b12ca76

**Pull request merged**